### DISCUSSION TITLE
About P1P2 energy consumption data

### DISCUSSION THREAD

**Discussion by kervanaslan** [Arnold-n/P1P2MQTT#36] [Q&A]
Hi Arnold, 

Firstly, thanks for this great job.
I'm wondering if it is possible to periodically get Energy consumption data from P1P2. Does P1P2 some how send data?

You defined KLICK-DA is a way to get it, but without KLICK-DA are we able to get it?

**Comment by Arnold-n** [comment]
Hi @kervanaslan,

Thanks for asking and your appreciation. This is possible on some systems using the "C2" command. Given limited resolution, an external meter is still preferred (and will be supported as input  over MQTT for COP calculation in the future).

The KLIC-DA is indeed not the preferred solution. It was a difficult hack made (with lots of trial and error) for a KLIC-DA owner where KLIC-DA was already acting as an auxiliary controller. There is little opportunity in the P1/P2 protocol to request the energy consumption counters. The KLIC-DA solution may not work reliably on other systems. And the KLIC-DA device ignores the heat pump replies itself, just as the main controller - the main controller only updates the counters if it requests the counters itself, and the KLIC-DA device relies on the main controller only.

Unfortunately within the Daikin protocol implementation there is no known "legal" way to request these counters from the heat pump - the main controller (the wall thermostat) only requests the counters when requested  manually, or at the end of each month. There is no (documented) way to trigger the main controller to request these counters. The current solution (enabled by the "C2" command) is to let the P1P2-ESP-interface request the counters directly from the heat pump. This may confuse the main controller; perhaps this leads in some systems to the main controller and the heat pump both responding at the same time. In my system the heat pump answers, and shortly after the main controller continues regular operation, so it works find. You can try "C2" and see if the communication pattern is acceptable. If not, it will result in errors, and after 10 errors the interface will switch itself of as a safety mechanism.

If the mechanism works (and if the heat pump does measure energy consumption and similar counters, your system may be different) the following information may be seen on MQTT topics P1P2/P/xxx/C/1/#:
```
P1P2/P/154/C/1/Electricity_Consumed_Backup_Heating 0
P1P2/P/154/C/1/Electricity_Consumed_Backup_DHW 0
P1P2/P/154/C/1/Electricity_Consumed_Compressor_Heating 16777
P1P2/P/154/C/1/Electricity_Consumed_Compressor_Cooling 0
P1P2/P/154/C/1/Electricity_Consumed_Compressor_DHW 0
P1P2/P/154/C/1/Electricity_Consumed_Total 39305
P1P2/P/154/C/1/Energy_Produced_Heatpump_Heating 148206
P1P2/P/154/C/1/Energy_Produced_Heatpump_Cooling 0
P1P2/P/154/C/1/Energy_Produced_Heatpump_DHW 0
P1P2/P/154/C/1/Energy_Produced_Heatpump_Total 148206
P1P2/P/154/C/1/Hours_Circulation_Pump 36037
P1P2/P/154/C/1/Hours_Compressor_Heating 28968
P1P2/P/154/C/1/Hours_Compressor_Cooling 0
P1P2/P/154/C/1/Hours_Compressor_DHW 0
P1P2/P/154/C/1/Hours_Backup1_Heating 0
P1P2/P/154/C/1/Hours_Backup1_DHW 0
P1P2/P/154/C/1/Hours_Backup2_Heating 0
P1P2/P/154/C/1/Hours_Backup2_DHW 0
P1P2/P/154/C/1/Hours_Unknown_34 0
P1P2/P/154/C/1/Hours_Unknown_35 0
P1P2/P/154/C/1/Counter_Unknown_00 0
P1P2/P/154/C/1/Counter_Unknown_01 0
P1P2/P/154/C/1/Counter_Unknown_02 0
P1P2/P/154/C/1/Starts_Compressor 21022
P1P2/P/154/C/1/Hours_Gasboiler_Heating 12857
P1P2/P/154/C/1/Hours_Gasboiler_DHW 513
P1P2/P/154/C/1/Counter_Gasboiler_Heating 0 (some systems report gas usage here)
P1P2/P/154/C/1/Counter_Gasboiler_DHW 0 (some systems report gas usage here)
P1P2/P/154/C/1/Starts_Gasboiler 46897
```
Strangely, the counter for Electricity_Consumed_Compressor_Heating stopped counting in 2018, also on the main controller from Daikin, even before I started with P1/P2 experiments, but the Electricity_Consumed_Total continues to update. There are other users with the same problem.

**Reply by kervanaslan** [reply]
`
The current solution (enabled by the "C2" command) is to let the P1P2-ESP-interface request the counters directly from the heat pump.
`

Hi Arnold,

Thanks for your response. While saying "Heat Pump" do you mean "VRV Indoor Unit for Heat Pump"? I couldn't understand it clearly what you mean.

**Reply by Arnold-n** [reply]
Sorry for not seeing your follow-up question. Actually I meant the Altherma indoor unit. For VRV/Sky units it is currently unknown whether energy consumption data is available via P1/P2.

**Comment by javimurcia** [comment]
A little insight onto new units (Daikin Altherma 3 R W): When you are not using the smart grid interface, you can connect 2 energy meters (via its pulse output) to the interior unit.

In that case, the meter reported values would be used, instead of the assumed.

If you use the smart grid function, one of the pulse inputs is used for the SG1 and SG2, and the second meter input is now the smart grid meter (the meter measuring the solar production, which will limite the maximum power of the system)
